### PR TITLE
PCG Updates

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR introduces a [new page](https://deploy-preview-2263--docs-spectrocloud.netlify.app/clusters/pcg/manage-pcg/pcg-upgrade) that explains how PCG upgrades work and what actions a user needs to take. Other changes include:

- Removed outdated [PCG troubleshooting steps](https://deploy-preview-2263--docs-spectrocloud.netlify.app/troubleshooting/pcg).
- Added a [section](https://deploy-preview-2263--docs-spectrocloud.netlify.app/clusters/pcg/architecture#operating-system) on the architecture page that discusses the underlying OS.
- Added a [section](https://deploy-preview-2263--docs-spectrocloud.netlify.app/clusters/pcg/architecture#network-proxy) on the architecture page that discusses network proxy.

## Review Changes

💻 [Preview URL]()

🎫 [DOC-1079](https://spectrocloud.atlassian.net/browse/DOC-1079)

[DOC-1079]: https://spectrocloud.atlassian.net/browse/DOC-1079?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/deprecated/troubleshooting/pcg.md
Comment (karl-cardenas-coding) [inline docs/deprecated/troubleshooting/pcg.md:1]: This content is deprecated, so no need to review this file

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+description: "Learn how to upgrade a Private Cloud Gateway (PCG) in Palette."
+hide_table_of_contents: false
+sidebar_position: 40
+tags: ["pcg"]
+---
+
+A PCG upgrade event can be broken down into two categories: Palette agent updates and cluster profile updates. Depending
+on the upgrade event, different steps are required to complete the upgrade.
+
+Palette updates are tied to the Palette platform version. Self-hosted Palette instances must upgrade to the next minor
```
Comment (lennessyy) [inline docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:13]:
```suggestion
Palette agent updates are tied to the Palette platform version. Self-hosted Palette instances must upgrade to the next minor
```
Is this correct? It makes sense for us to discuss the two update versions directly after introducing them so I assume that Palette update here is referring to Palette agent updates. Also, since we have a lot of concepts that sounds similar but are different in this section, it may be best to explicitly say that PCG agent and Palette Agent are the same or just standardize and only use Palette Agent,

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Palette. For example, if you are on Palette 4.1.12, you must upgrade to Palette 4.2.0 or greater to receive the latest
+Palette agent updates that are compatible with Palette 4.2.x.
+
+:::warning
+
+Before upgrading a self-hosted Palette instance to a new minor or major version, allow the PCG agent to automatically
+upgrade to the latest version and manually approve any pending cluster profile updates. This will help you avoid
+compatibility issues between the PCG and the upgraded Palette.
+
+For instance, if you are upgrading from Palette v3.4.x to v4.2.x, ensure both PCG agent and cluster profiles updates are
```
Comment (lennessyy) [inline docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:24]: This confused me a bit - maybe I am missing some context. When you upgrade Palette from 3.4 to 4.2, do you need to upgrade incrementally? For example, 3.4 -> 4.0 first and then 4.0 -> 4.1 and then 4.1 -> 4.2. It kind of reads like this is the case. 

If so, I think it would help to provide that context here as maybe readers aren't familiar with the Palette. And then clarify what we mean is that the user should make sure that the PCG agent is the most up-to-date with their current minor version

**Review by lennessyy — APPROVED** [review]

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+| --------------- | --------------------- | ------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Palette Agent   | No                    | No                 | Automatically updated unless platform updates or cluster updates are paused. Refer to the [Pause Platform Upgrades](../../cluster-management/platform-settings/pause-platform-upgrades.md) to learn more about pausing updates.                                                                                |
+| Cluster Profile | Yes                   | Potentially        | Manually approve the cluster profile update to apply the latest cluster profile changes. Refer to the [Pause Platform Upgrades](../../cluster-management/platform-settings/pause-platform-upgrades.md) to learn more about pausing updates. Not all Palette version updates introduce cluster profile changes. |
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:24]: "Potential Expected Downtime?" I'm not aware of potential downtime. New machines are supposed to be provisioned, and the PCG components will be moved there.

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+---
+sidebar_label: "Upgrade a PCG"
+title: "Upgrade a PCG"
+description: "Learn how to upgrade a Private Cloud Gateway (PCG) in Palette."
+hide_table_of_contents: false
+sidebar_position: 40
+tags: ["pcg"]
+---
+
+A PCG upgrade event can be broken down into two categories: Palette agent updates and cluster profile updates. Depending
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:10]: "cluster profile updates" > it's more a "hidden" cluster profile, specific to the PCG type (depending on the cloud). I see it's mentioned below in the page, but I think we should also mention a quick sentence at the top to avoid someone reading through more than 50% of the page to discover this concept; just my 2c.

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+---
+sidebar_label: "Upgrade a PCG"
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:2]: Is this page relevant to all PCGs types (the 4 types)?

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md (hunk 3 above)
Comment (cloudmaniac) [inline docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:24]:
```suggestion
| Cluster Profile | Yes                   | Potentially        | Manually approve the cluster profile update to apply the latest cluster profile changes. Not all Palette version updates introduce cluster profile changes. |
```

This specific sentence doesn't apply to this component.

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+The cluster profile for a PCG is locked down and cannot be changed by a user. In the scenario when a cluster profile is
+update available, for example, when upgrading the underlying Kubernetes version of the PCG cluster, the PCG cluster
+detail's page will display a notification that a new version of the cluster profile is available.
+
+    ![A PCG details page displaying an eligble update](/clusters_manage-pcg_pcg-upgrade_updates-button.png)
+
+To upgrade the PCG cluster, click the **Updates** button to start the upgrade process. A cluster profile editor will
+appear, displaying the changes that will be made to the cluster profile. Review the changes and click **Confirm
+updates** to start the upgrade process. Depending on the changes, the upgrade process may take some time to complete,
+and the PCG cluster may be unavailable during the upgrade process. Refer to the
```
Comment (cloudmaniac) [inline docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:58]:
```suggestion
and the PCG cluster may be unavailable during the upgrade process. 
```

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md (hunk 4 above)
Comment (cloudmaniac) [reply docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:10]: Maybe the page should use "PCG cluster profile" in all occurrences as well?

**Review by cloudmaniac — COMMENTED** [review]
Only a few suggestions. Otherwise LGTM.

File: docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md (hunk 3 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/pcg/manage-pcg/pcg-upgrade.md:24]: I updated the heading to state PCG Downtime as the context is only for the PCG. Great call out 👍🏻
